### PR TITLE
UG improvements

### PR DESCRIPTION
**Pull request by Vatinius**
Resolves issues #315, #300, #260, #176, #329, #330 assigned to me.

Will assign reviewers when PR is ready for review.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -690,3 +679,15 @@ Furthermore, certain edits can cause the PROperty to behave in unexpected ways (
 - **CONTRACTOR**: Professionals providing construction, renovation, or repair services.
 
 [_Back to Top_](#table-of-contents)
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Appendix
+
+### Planned Enhancements
```
Comment (Tkaixiang) [inline docs/UserGuide.md:687]: I think we were planning to also add #315 (improved error handling for export) into future enhancements?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -551,12 +546,6 @@ Example:
 
 1. `help` results in a popup window which contains all the commands in PROperty for easy reference during use.
 
-Visual example of correct output:
-
-![HelpCommandShowcase.png](images/user-guide-images/HelpMessageShowcase.png)
-![help message](images/user-guide-images/HelpMessageShowcase.png)
-![help message](images/user-guide-images/HelpMessageShowcase2.png)
-
```
Comment (Tkaixiang) [inline docs/UserGuide.md:559]: I think we should remove these unused images :)

File: docs/UserGuide.md [hunk 3]
```diff
@@ -510,21 +506,19 @@ Exports your contacts and their relevant data to a Comma-Separated Value (CSV) f
...
 </div>
+
 Example:
 
 1. `export` exports your current client data in PROperty into a CSV file.
 
-Visual example of correct output [Example `1.`]:
-
-![ExportCommandShowcase](images/user-guide-images/ExportCommandShowcase.png)
-
```
Comment (Tkaixiang) [inline docs/UserGuide.md:527]: I think we should remove these unused images :)

File: docs/UserGuide.md [hunk 4]
```diff
@@ -408,10 +408,6 @@ Example:
 
 1. `clear` clears all clients and their data from PROperty.
 
-Visual example of correct output [Example `1.`]:
-
-![ClearCommandShowcase.png](images/user-guide-images/ClearCommandShowcase.png)
-
```
Comment (Tkaixiang) [inline docs/UserGuide.md:414]: I think we should remove these unused images :)

File: docs/UserGuide.md (hunk 1 above)
Comment (Vatinius) [reply docs/UserGuide.md:687]: I actually addressed #315 under the export command. Do you feel it belongs here instead?

File: docs/UserGuide.md (hunk 2 above)
Comment (Vatinius) [reply docs/UserGuide.md:559]: Alright sure! Will get to t.

File: docs/UserGuide.md (hunk 1 above)
Comment (Tkaixiang) [reply docs/UserGuide.md]: Discussed to add it into planned enhancements still on tele, pending #331 to be merged

**Review by Tkaixiang — APPROVED** [review]
pre-approving as all of my comments have been addressed, pending the movement of Future Enhancements here into the DG

**Comment by Vatinius** [comment]
> pre-approving as all of my comments have been addressed, pending the movement of Future Enhancements here into the DG

Do take a look now!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -968,6 +968,20 @@ separating sections of a phone number (e.g., `+1-800-9087-2029`).
...
+(e.g. using `find 29` to find a client with the address `Blk 30 Geylang Street 29, #06-40` will result in no clients being found).
+
+**Proposed Enhancement**
+
+We plan to enhance the search feature to allow partial matches on numeric values regardless of punctuation or spacing. 
+This enhancement will improve the user experience by showing relevant contact details even if the input includes numbers adjacent to punctuation marks.
+
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Appendix C: Instructions for Manual Testing**
```
Comment (Tkaixiang) [inline docs/DeveloperGuide.md:987]: Hmm, what about the export future enhancement?

Comment (Vatinius) [reply docs/DeveloperGuide.md:987]: Ahh forgot about that! Thanks for reminding ahahah

Comment (Vatinius) [reply docs/DeveloperGuide.md:1003]: Attended to!

**Pull request merged**